packages/python/plotly/plotly/express/_core.py

```diff
@@ -1031,6 +1064,32 @@ def build_dataframe(args, attrables, array_attrables):
                 args[field_name] = str(col_name)
             else:
                 args[field_name][i] = str(col_name)
+            wide_id_vars.add(str(col_name))
+
+    if wide_mode:
+        # TODO multi-level index
+        # TODO multi-level columns
+        index_name = df_output.index.name or "index"
+        wide_id_vars.add(index_name)
+        if index_name not in df_output.columns:
```

Comment:
so if there is a column with name "index" it will be used for `x`, right?

Reply:
because otherwise this statement is always false I think

Reply:
If something like `hover_name=df.index` happens this is needed to prevent the `melt()` from failing.